kaldb/src/main/java/com/slack/kaldb/logstore/search/aggregations/AggBuilderBase.java

```diff
@@ -0,0 +1,42 @@
+package com.slack.kaldb.logstore.search.aggregations;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Provides common functionality for all aggregations. All aggregations are expected to extend this
+ * class, or another base class that extends this one (ie, ValueSourceAggBuilder).
+ */
+public abstract class AggBuilderBase implements AggBuilder {
+  // This is the name of the aggregation, as provided by the user. In most cases this will be a
+  // unique string ID to correlate the output values - Grafana generally uses incrementing
+  // integers as a string (ie "1", "2").
+  protected final String name;
+  protected final Map<String, Object> metadata;
```

Comment:
is the `metadata` used anywhere?

Reply:
Not yet - but this is implemented in OpenSearch `AbstractAggregationBuilder` and can be passed to the aggregator builders, so I went ahead and stubbed it out. I'm not sure if we'll eventually need it, so it's currently just implemented as empty maps for now.